File: prime_numbers/benchmark.rb
def get_primes7(n)
    if n < 2
        return []
    end
     
    if n == 2
        return [2]
    end
     
    #s = Range.new(3, n, 2)???
    s = []
    i = 3
    while i < n + 1
        s.push(i)
        i += 2
    end
     
    mroot = Math.sqrt(n)
 
    half = s.length
    i = 0
    m = 3
    while m <= mroot
        if s[i] != 0
            j = (m * m - 3) / 2
            s[j] = 0
            while j < half
                s[j] = 0
                j += m
            end
        end
        i = i + 1
        m = 2 * i + 3
    end
     
    res = [2]
    for v in s
        if v != 0
            res.push(v);
        end
    end
#   x = 0
#   while x &lt; s.length
#       if s[x] != 0
#           res.push(s[x])
#       end
#       x = x + 1
#   end
    return res
end
 
 
res = [];
start = Time.now
for i in 1..10
    res = get_primes7 10000000
    puts "Found #{res.length} prime numbers.\n";
end
elapsed = Time.now - start
puts "Execution time: #{elapsed}.\n"

File: prime_numbers/benchmark.go
package main

import (
	"fmt"
	"math"
	"time"
)

func getPrimes7(n int) []int {
	if n < 2 {
		return []int{}
	}
	if n == 2 {
		return []int{2}
	}

	s := []int{}
	for i := 3; i <= n; i += 2 {
		s = append(s, i)
	}

	var j int
	m := 3
	mroot := int(math.Sqrt(float64(n)))
	half := len(s)
	for i := 0; m <= mroot; {
		if s[i] != 0 {
			j = (m*m - 3) / 2
			s[j] = 0
			for j < half {
				s[j] = 0
				j += m
			}
		}
		i++
		m = 2*i + 3
	}

	res := []int{}
	res = append(res, 2)
	for _, v := range s {
		if v != 0 {
			res = append(res, v)
		}
	}

	return res
}

func main() {
	var startTime = time.Now()

	var res []int

	for i := 0; i < 10; i++ {
		res = getPrimes7(10000000)
		fmt.Printf("Found %d prime numbers.\n", len(res))
	}

	var duration = time.Since(startTime)
	fmt.Println("Execution time: ", duration.Seconds())
}
